### ISSUE TITLE
Nextcloud android crash

### ISSUE THREAD

**Issue by sonic1200** [nextcloud/android#4671]
### Actual behaviour
- Tell us what happens
Application crash when clicking on a notification dismiss

### Expected behaviour
- Tell us what should happen
Continue to main screen of nextcloud app
 
### Steps to reproduce
1.  Open notification screen
2.  Click on 'notify later' (deny) 
3. App crash

### Environment data
Android version:

Device model: 

Stock or customized system:

Nextcloud app version:

Nextcloud server version:

#### Web server error log
```
Insert your webserver log here
```

#### Nextcloud log (data/nextcloud.log)
```
Insert your Nextcloud log here
```
**NOTE:** Be super sure to remove sensitive data like passwords, note that everybody can look here! You can use the Issue Template application to prefill some of the required information: https://apps.nextcloud.com/apps/issuetemplate

************ CAUSE OF ERROR ************

java.lang.IllegalStateException: Cannot execute task: the task has already been executed (a task can be executed only once)
	at android.os.AsyncTask.executeOnExecutor(AsyncTask.java:640)
	at android.os.AsyncTask.execute(AsyncTask.java:595)
	at com.owncloud.android.ui.adapter.NotificationListAdapter.lambda$onBindViewHolder$2$NotificationListAdapter(NotificationListAdapter.java:182)
	at com.owncloud.android.ui.adapter.-$$Lambda$NotificationListAdapter$TXM0JwRxGeq8tl4SIWzikYjFQ-k.onClick(Unknown Source:8)
	at android.view.View.performClick(View.java:7352)
	at android.widget.TextView.performClick(TextView.java:14177)
	at android.view.View.performClickInternal(View.java:7318)
	at android.view.View.access$3200(View.java:846)
	at android.view.View$PerformClick.run(View.java:27800)
	at android.os.Handler.handleCallback(Handler.java:873)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at android.os.Looper.loop(Looper.java:214)
	at android.app.ActivityThread.main(ActivityThread.java:7050)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:964)

************ APP INFORMATION ************
ID: com.nextcloud.client
Version: 30080190
Build flavor: gplay

************ DEVICE INFORMATION ************
Brand: samsung
Device: dreamqltecan
Model: SM-G950W
Id: PPR1.180610.011
Product: dreamqltevl

************ FIRMWARE ************
SDK: 28
Release: 9
Incremental: G950WVLS6CSH1

**Comment by ezaquarii** [comment]
@AndyScherzinger I checked the code and it looks like we explicitly tried to re-use the async task.
Do you remember why we did 941239c19ade47b076bb4a114b78be0fb82c3fc4 in the first place?

**Comment by AndyScherzinger** [comment]
Don't remember why we did it, unfortunately. Maybe @tobiasKaminsky can remember it?

**Comment by AndyScherzinger** [comment]
Cc @tobiasKaminsky

**Comment by AndyScherzinger** [comment]
Cc @tobiasKaminsky

**Comment by tobiasKaminsky** [comment]
> @AndyScherzinger I checked the code and it looks like we explicitly tried to re-use the async task.
> Do you remember why we did [941239c](https://github.com/nextcloud/android/commit/941239c19ade47b076bb4a114b78be0fb82c3fc4) in the first place?

I guess this was due to codacy/CI saying that this should not be generated within a loop.
But this is wrong, as we have to create a new task on every click / for every action.

I will fix it.
